app/src/main/java/com/example/woloxapp/ui/login/LoginViewModel.kt

```diff
@@ -49,27 +59,28 @@ class LoginViewModel(private val app: Application) : AndroidViewModel(app) {
         }
         _userIsLogged.value = savedEmail != null && savedPassword != null
     }
-    private val userRepository = UserRepository()
+
     fun login(user: User) {
         viewModelScope.launch {
-            val response = userRepository.loginUser(user)
-            if (response.isSuccessful) {
-                val editor = sharedPreferencesSaved.edit()
-                editor.also {
-                    it.putString(USERNAME, user.email)
-                    it.putString(PASSWORD, user.password)
-                    it.putString(DATA_USER, Gson().toJson(response.body()))
-                    it.commit()
+            _response.value = null
+            userRepository.login(user).collect {
+                    values ->
+                _response.value = values
+                when (response.value) {
+                    is NetworkResult.Success<*> -> {
+                        editor.also {
+                            it.putString(DATA_USER, Gson().toJson(response.value))
```

Comment:
You don't need parse to json here. This will generate error when you try to access any property. can you use the user model instead? for example: "response.value.{anyProperty}" and save it in preferences.